### PR TITLE
PMM-10918 Add backups  and restores for DBaaS

### PR DESCRIPTION
**Pull request by gen1us2k**
PMM-10918

Build: SUBMODULES-0

- [ ] Links to other linked pull requests (optional).

### CODE REVIEW

File: managed/services/dbaas/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -345,6 +345,12 @@ func (k *Kubernetes) CreatePMMSecret(secretName string, secrets map[string][]byt
 	return k.client.ApplyObject(secret)
 }
 
+func (k *Kubernetes) CreateRestore(restore *dbaasv1.DatabaseClusterRestore) error {
+	k.lock.Lock()
+	defer k.lock.Unlock()
+	return k.client.ApplyObject(restore)
+}
```
Comment (recharte) [inline managed/services/dbaas/kubernetes/kubernetes.go:352]: Thank you dbaas-operator for the simplicity you bring to PMM :pray:

File: managed/services/dbaas/kubernetes/types.go [hunk 2]
```diff
@@ -162,22 +169,48 @@ func DatabaseClusterForPXC(cluster *dbaasv1beta1.CreatePXCClusterRequest, cluste
...
-			return nil, err
+			return nil, nil, err
 		}
 		dbCluster.Spec.LoadBalancer.Size = cluster.Params.ClusterSize
 		dbCluster.Spec.LoadBalancer.Image = cluster.Params.Proxysql.Image
 		dbCluster.Spec.LoadBalancer.Resources = resources
 		dbCluster.Spec.LoadBalancer.Type = "proxysql"
 	}
+	if cluster.Params.Backup != nil {
+
```
Comment (recharte) [inline managed/services/dbaas/kubernetes/types.go:180]:
```suggestion

```

File: api/managementpb/dbaas/pxc_clusters.proto [hunk 3]
```diff
@@ -20,6 +20,9 @@ enum PXCBackupState {
   PXC_BACKUP_STATE_FAILED = 3;
 }
 
+
+
+
```
Comment (recharte) [inline api/managementpb/dbaas/pxc_clusters.proto:25]: A bit too much space :see_no_evil: 
```suggestion
```

File: managed/services/dbaas/kubernetes/types.go [hunk 4]
```diff
@@ -241,18 +321,47 @@ func DatabaseClusterForPSMDB(cluster *dbaasv1beta1.CreatePSMDBClusterRequest, cl
...
-			return dbCluster, fmt.Errorf("failed to recognize expose annotations for %s cluster type", clusterType)
+			return dbCluster, nil, fmt.Errorf("failed to recognize expose annotations for %s cluster type", clusterType)
 		}
 		dbCluster.Spec.LoadBalancer.Annotations = annotations
 		if cluster.InternetFacing && clusterType == ClusterTypeEKS {
 			dbCluster.Spec.LoadBalancer.Annotations["service.beta.kubernetes.io/aws-load-balancer-type"] = "external"
 		}
 	}
+	if cluster.Params.Backup != nil {
+
```
Comment (recharte) [inline managed/services/dbaas/kubernetes/types.go:337]:
```suggestion
```

File: managed/services/management/dbaas/db_cluster_service.go [hunk 5]
```diff
@@ -369,6 +373,45 @@ func (s DBClusterService) DeleteDBCluster(ctx context.Context, req *dbaasv1beta1
 	return &dbaasv1beta1.DeleteDBClusterResponse{}, nil
 }
 
+// ListS3Backups returns list of backup artifacts stored on s3
+func (s DBClusterService) ListS3Backups(ctx context.Context, req *dbaasv1beta1.ListS3BackupsRequest) (*dbaasv1beta1.ListS3BackupsResponse, error) {
+	if req == nil && (req != nil && req.LocationId == "") {
```
Comment (recharte) [inline managed/services/management/dbaas/db_cluster_service.go:378]:
```suggestion
	if req == nil || (req != nil && req.LocationId == "") {
```

**Review by percona-csalguero — APPROVED** [review]

**Review by recharte — APPROVED** [review]

**Pull request merged**